go/embedded/online_features.go

```diff
@@ -0,0 +1,113 @@
+package embedded
+
+import (
+	"context"
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/apache/arrow/go/arrow/cdata"
+	"github.com/feast-dev/feast/go/internal/feast"
+	"github.com/feast-dev/feast/go/protos/feast/types"
+	"github.com/feast-dev/feast/go/utils"
+	"log"
+)
+
+type OnlineFeatureService struct {
+	fs *feast.FeatureStore
+}
+
+type OnlineFeatureServiceConfig struct {
+	RepoPath   string
+	RepoConfig string
+}
+
+type DataTable struct {
+	DataPtr   uintptr
+	SchemaPtr uintptr
+}
+
+func NewOnlineFeatureService(conf *OnlineFeatureServiceConfig) *OnlineFeatureService {
+	repoConfig, err := feast.NewRepoConfigFromJSON(conf.RepoPath, conf.RepoConfig)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	fs, err := feast.NewFeatureStore(repoConfig)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	defer fs.DestructOnlineStore()
+	return &OnlineFeatureService{fs: fs}
+}
+
+func (s *OnlineFeatureService) GetOnlineFeatures(
+	featureRefs []string,
+	featureServiceName string,
+	entities DataTable,
+	fullFeatureNames bool,
+	projectName string,
+	output DataTable) error {
+
+	entitiesRecord, err := readArrowRecord(entities)
+	if err != nil {
+		return err
+	}
+
+	numRows := entitiesRecord.Column(0).Len()
+
+	entitiesProto, err := recordToProto(entitiesRecord)
+	if err != nil {
+		return err
+	}
+
+	var featureService *feast.FeatureService
+	if featureServiceName != "" {
+		featureService, err = s.fs.GetFeatureService(featureServiceName, projectName)
+	}
+
+	resp, err := s.fs.GetOnlineFeatures(
```

Comment:
I guess this has nothing to do with your PR, but do you think it may be more natural for these internal functions to be `fs.GetOnlinFeaturesByFeatureService()` and `fs.GetOnlineFeaturesByFeatureRefs()`?